File: app/src/main/java/com/wipro/codingexcercise/model/FactsDto.java
package com.wipro.codingexcercise.model;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import java.util.ArrayList;


/**
 * @author jayeshhiray
 * THis is the main pojo class we get from the API call
 */
public class FactsDto implements Parcelable {

    public static final Creator<FactsDto> CREATOR = new Creator<FactsDto>() {
        @Override
        public FactsDto createFromParcel(Parcel source) {
            return new FactsDto(source);
        }

        @Override
        public FactsDto[] newArray(int size) {
            return new FactsDto[size];
        }
    };
    private String title;
    private ArrayList<Rows> rows;

    public FactsDto() {
    }

    private FactsDto(Parcel in) {
        this.title = in.readString();
        this.rows = new ArrayList<>();
        in.readList(this.rows, Rows.class.getClassLoader());
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public ArrayList<Rows> getRows() {
        return rows;
    }

    public void setRows(ArrayList<Rows> rows) {
        this.rows = rows;
    }

    @NonNull
    @Override
    public String toString() {
        return "ClassPojo [title = " + title + ", rows = " + rows + "]";
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.title);
        dest.writeList(this.rows);
    }
}

File: app/src/main/java/com/wipro/codingexcercise/model/FactsDto.kt
package com.wipro.codingexcercise.model

import android.os.Parcel
import android.os.Parcelable
import java.util.*

/**
 * @author jayeshhiray
 * THis is the main pojo class we get from the API call
 */
class FactsDto : Parcelable {
    var title: String? = null
    var rows: ArrayList<Rows?>? = null

    constructor() {}
    private constructor(`in`: Parcel) {
        title = `in`.readString()
        rows = ArrayList()
        `in`.readList(rows, Rows::class.java.classLoader)
    }

    override fun toString(): String {
        return "ClassPojo [title = $title, rows = $rows]"
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeList(rows)
    }

    companion object {
        val CREATOR: Parcelable.Creator<FactsDto?> = object : Parcelable.Creator<FactsDto?> {
            override fun createFromParcel(source: Parcel): FactsDto? {
                return FactsDto(source)
            }

            override fun newArray(size: Int): Array<FactsDto?> {
                return arrayOfNulls(size)
            }
        }
    }
}